service/controller/resource/tccpn/create.go

```diff
@@ -288,15 +288,17 @@ func (r *Resource) newAutoScalingGroup(ctx context.Context, cr infrastructurev1a
 		HAMasters: haMastersEnabled,
 	}
 	for _, m := range mappings {
+
+		dependsOn := []string{key.ControlPlaneENIResourceName(m.ID), key.ControlPlaneVolumeResourceName(m.ID)}
+		// ASG for second and third master will have chain dependency on the previous one
+		// to have rolling update of one ASG after the previous one.
+		if m.ID == 2 || m.ID == 3 {
+			dependsOn = append(dependsOn, key.ControlPlaneASGResourceName(&cr, m.ID-1))
```

Comment:
maybe there is a better way how to do this?

Reply:
I find that pretty smart actually. It is a bit odd but I have no better idea. 😄